clients/src/main/java/org/apache/kafka/common/security/authenticator/SaslServerAuthenticator.java

```diff
@@ -73,6 +73,7 @@
 
 public class SaslServerAuthenticator implements Authenticator {
 
+    private static final int MAX_RECEIVE_SIZE = 131072; // Chosen as twice the max receive size for GSSAPI
```

Comment:
By the way, it's not necessary to get a super tight bound here; we're just trying to provide a safer default out of the box so that at least a misconfigured non-kafka client connecting to the broker won't cause a giant allocation. To allow for custom sasl methods, I think we may want to bump this a little higher. Maybe 512K is reasonable?